olp-cpp-sdk-dataservice-read/src/repositories/DataRepository.cpp

```diff
@@ -149,6 +136,14 @@ DataResponse DataRepository::GetBlobData(
     return {{client::ErrorCode::PreconditionFailed, "Data handle is missing"}};
   }
 
+  NamedMutex mutex(catalog.ToString() + *data_handle);
+  std::unique_lock<NamedMutex> lock(mutex, std::defer_lock);
+
```

Comment:
Maybe add layer to the mutex name.

Reply:
Good point. According to DS documentation, the data handle is only to be unique per layer.

Reply:
Added layer